src/kaleidoscope/device/avr/ATMega32U4.h

```diff
@@ -0,0 +1,51 @@
+/* -*- mode: c++ -*-
+ * device::avr::ATMega32U4 -- Generic ATMega32U4 device base class
```

Comment:
In context, device::avr::ATMega32U4 reads really funny. it looks like it's the name of the microcontroller. 
I think we want to call it something like ATMega32U4Keyboard. And while it might subclass something like an AVRKeyboard, I don't -think- we want it inside an 'avr' namespace.

Reply:
`device::ATMega32U4Device`? I don't want to call it "keyboard"... but Device is redundant, and we do not have support for non-keyboard devices yet, so... keyboard will do for the time being.